pkg/operator/operator.go

```diff
@@ -489,6 +501,16 @@ func (o *Operator) sync(key string) error {
 	return nil
 }
 
+func (o *Operator) reportError(err error, taskName string) error {
+	if o.failedReconcileAttempts < 3 {
+		o.failedReconcileAttempts++
+		return nil
+	}
```

Comment:
should we reset `o.failedReconcileAttempts`?

Reply:
No, I think we only want to do reset once we were successful. This happens [here](https://github.com/openshift/cluster-monitoring-operator/pull/1193/files#diff-99c0290799daf9abc6240df64063e20bfaf67b371577b67ac7eec6f4725622ffR495)

Reply:
So the operator would retry up to 3 times before reporting available=false/degraded=true and the next sync() operations wouldn't retry?

```
Updating ClusterOperator status to failed after retrying 3 times. Err: xxx
...
Updating ClusterOperator status to failed after retrying 4 times. Err: xxx
...
Updating ClusterOperator status to failed after retrying 5 times. Err: xxx
```

Reply:
The intention is that it allows for 3 failed attempts before reporting `available=false/degraded=true`. All error cause a retry to sync the event in question though. `o.failedReconcileAttempts` is really only affecting if/when the operator status changes...unless I'm missing something.

The status and `o.failedReconcileAttempts` is reset to `0` and `available=true/degraded=false` respectively once a call to sync completes without error

Reply:
:+1: